onboarding/source/amb_light_service.c

```diff
@@ -18,22 +29,113 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if (lightServiceTaskHandle == NULL) {
+        // Create light service task
+        xReturned = xTaskCreate(lightServiceTask,
+                                LIGHT_SERVICE_NAME,
+                                LIGHT_SERVICE_STACK_SIZE,
+                                NULL,
+                                LIGHT_SERVICE_PRIORITY,
+                                &lightServiceTaskHandle);
+    }
+
+    // Check if task was created successfully
+    if (xReturned == pdFAIL) {
+        return 0;
+    }
 
+    if (lightServiceQueue == NULL) {
+        // Create light service queue
+        lightServiceQueue = xQueueCreate(LIGHT_SERVICE_QUEUE_LENGTH,
+                                        LIGHT_SERVICE_QUEUE_ITEM_SIZE);
+    }
+
+    // Check if queue was created successfully
+    if (lightServiceQueue == NULL) {
+        return 0;
+    }
     /* USER CODE END */
     return 1;
 }
 
+static void adcStartConversion_selChn(adcBASE_t *adc, unsigned channel, unsigned fifo_size, unsigned group) {
+    /** - Setup FiFo size */
+    adc->GxINTCR[group] = fifo_size;
+
+    /** - Start Conversion */
+    adc->GxSEL[group] = 1 << channel;
+}
+
+static void adcGetSingleData(adcBASE_t *adc, unsigned group, adcData_t *data) {
+    unsigned  buf;
+    adcData_t *ptr = data; 
+
+    /** -  Get conversion data and channel/pin id */
+    buf        = adc->GxBUF[group].BUF0;
+    ptr->value = (unsigned short)(buf & 0xFFFU);
+    ptr->id    = (unsigned short)((buf >> 16U) & 0x1FU); // int to unsigned short
+
+    adc->GxINTFLG[group] = 9U;
+}
+
+static uint32 getLightSensorData(void) {
+	adcData_t adc_data;
+	adcData_t *adc_data_ptr = &adc_data;
+
+ 	/** - Start Group1 ADC Conversion 
+ 	*     Select Channel 6 - Light Sensor for Conversion
+ 	*/
+	adcStartConversion_selChn(adcREG1, LIGHT_SENSOR, 1, adcGROUP1);
+
+ 	/** - Wait for ADC Group1 conversion to complete */
+ 	while(!adcIsConversionComplete(adcREG1, adcGROUP1)); 
+
+	/** - Read the conversion result
+	*     The data contains the Light sensor data
+    */
+	adcGetSingleData(adcREG1, adcGROUP1, adc_data_ptr);
+	
+	/** - Transmit the Conversion data to PC using SCI
+    */
+	return (adc_data_ptr->value);
+}
+
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+    ASSERT(lightServiceQueue != NULL);
+
+    light_event_t *light_event;
+    BaseType_t returned = xQueueReceive(lightServiceQueue,
+                                        &light_event,
+                                        (TickType_t) 0);
 
+    if (returned == pdPASS) {
+        uint32 sensor_data = getLightSensorData();
```

Comment:
We typically use the fixed-size integer types with the `_t` suffix, such as `uint32_t`. They're both exactly the same, but it's good to keep things consistent.

Reply:
Make this change anywhere you use `uint32`. You'll need to make sure that `stdint.h` is included